### PR TITLE
commit progress #1644

### PR DESCRIPTION
**Pull request by salomon-j**

### CODE REVIEW

File: grails-app/controllers/au/org/ala/biocollect/BioActivityController.groovy [hunk 1]
```diff
@@ -1527,6 +1527,91 @@ class BioActivityController {
...
+            response.status = 401
+            model.error = "Access denied: User has not been authenticated."
+        } else if (!activity) {
+            model.error = "Invalid activity id"
+        } else if (!activity) {
+            model.error = "Invalid activity - ${id}"
+        } else if (!projectId) {
+            model.error = "No project associated with the activity"
+        } else if (projectService.isUserAdminForProject(userId, projectId) || activityService.isUserOwnerForActivity(userId, activity?.activityId)) {
+            model = [activity: activity, returnTo: params.returnTo, mode: mode]
```
Comment (temi) [inline grails-app/controllers/au/org/ala/biocollect/BioActivityController.groovy:1601]: returnTo and mode not required

File: grails-app/controllers/au/org/ala/biocollect/BioActivityController.groovy [hunk 2]
```diff
@@ -1527,6 +1527,91 @@ class BioActivityController {
...
+            def pActivity = projectActivityService.get(activity?.projectActivityId, "all")
```
Comment (temi) [inline grails-app/controllers/au/org/ala/biocollect/BioActivityController.groovy:1602]: project, pActivity, projectActivityId and id not required

File: src/main/groovy/au/org/ala/biocollect/swagger/model/BioCollectModels.groovy [hunk 3]
```diff
@@ -309,3 +309,13 @@ class ProjectSearchResponse {
     Integer total
     List<Facet> facets
 }
+
+// classes for "ws/bioactivity/data/simplified/{id}
+@JsonIgnoreProperties('metaClass')
+class GetOutputForActivitySimplifiedResponse {
+    Map activity
+    Map project
```
Comment (temi) [inline src/main/groovy/au/org/ala/biocollect/swagger/model/BioCollectModels.groovy:317]: make appropriate changes

**Review by temi — CHANGES_REQUESTED** [review]

File: src/main/groovy/au/org/ala/biocollect/swagger/model/BioCollectModels.groovy (hunk 3 above)
Comment (salomon-j) [reply src/main/groovy/au/org/ala/biocollect/swagger/model/BioCollectModels.groovy]: @temi I've now pushed the code review updates. thank you.

**Review by temi — APPROVED** [review]

**Comment by temi** [comment]
LGTM

**Pull request merged**